### DISCUSSION TITLE
Manage Terraform resources?

### DISCUSSION THREAD

**Discussion by varsis** [argoproj/argo-cd#10538] [Q&A]
Currently I have resources such as managed databases or pubsub running as part of my CI pipeline. This isn't desirable as it's a separate repo and requires it's own flow. 

Is it possible to have ArgoCD manage terraform resources similar to how it manages k8s resources? 

Thinking out loud I was thinking a presync hook but I'm not sure that's the best option. 

How are others managing non-k8s deployments?

**Comment by awx-fuyuanchu** [comment]
You can use [crossplane](https://crossplane.io) to manage non-k8s resources using k8s CRs.

**Comment by kostis-codefresh** [comment]
Apart from crossplane you should also check one of the several terraform controllers.
